### DISCUSSION TITLE
Running the Ethereum connector on L2s

### DISCUSSION THREAD

**Discussion by merklefruit** [getdozer/dozer#1454] [Q&A] [answered]
Hi! I'm experimenting with Dozer as it sounds really interesting for my use case. I have a few questions with regard to the Ethereum connector:

- how do I enable the ethereum feature on the binary installed from `brew`? Or do I have to install it locally specifying the cargo feature for it? Couldn't find any docs on the installation page on this
- I imagine the eth connector should work for all L2s that run on a `geth` fork, however has anyone done experiments on this and tested other networks such as Optimism or Arbitrum? I'm open to contributing to this if I choose to use Dozer :)

**Comment by chubei** [comment, marked as answer]
Hi thanks for your interest in Dozer!

Regarding the first question, currently the ethereum connector can only be used if you clone the repo and build from source. You can use the following commands:

```bash
git clone https://github.com/getdozer/dozer.git
cd dozer
git checkout v0.1.17 # This is the latest release for now. Or leave the repo on `main` branch if you'd like to live on the edge.
cargo build --bin dozer --features ethereum
```

**Comment by v3g42** [comment]
Regarding the second point, we haven't tested with L2s yet but in theory they should work. We have an internal sample pushing real-time ETH smart contract data on a grafana dashboard using a plug & play plugin :) 

Let us know how your experiment goes. We are happy to help. 

> * I imagine the eth connector should work for all L2s that run on a `geth` fork, however has anyone done experiments on this and tested other networks such as Optimism or Arbitrum? I'm open to contributing to this if I choose to use Dozer :)

**Comment by merklefruit** [comment]
Thank you to both for the extremely precise and fast replies!

**Comment by merklefruit** [comment]
Good news on the L2 side, I just tested with an Arbitrum archive node and it's working just fine!
Trying Optimism and Polygon soon.

While I am here, I have another question: it looks like the eth_traces table doesn't have a tx_hash or block_number field (or at least, I cannot find them), is there a reason for this? It would be useful for instance to record every transaction sent to a specific smart contract (unless there is already a way to do this, of which I am unaware). Ethereum-ETL traces are more complete as per [this spec](https://ethereum-etl.readthedocs.io/en/latest/schema/#tracescsv)

**Reply by chubei** [reply]
I'm not the one who implemented it but a quick glimpse over the code points me to [`struct Trace`](https://github.com/getdozer/dozer/blob/main/dozer-ingestion/src/connectors/ethereum/trace/helper.rs#L24).

I guess by including more fields and modifying [schema](https://github.com/getdozer/dozer/blob/main/dozer-ingestion/src/connectors/ethereum/trace/helper.rs#L93) and [mapping](https://github.com/getdozer/dozer/blob/main/dozer-ingestion/src/connectors/ethereum/trace/helper.rs#L150) from `Trace` to `Record`, we can expose `tx_hash` and `block_number`.

**Reply by merklefruit** [reply]
Thanks, I am looking into adding a filter to Traces to do what I need. Will close this discussion for now

**Reply by abhishekmishragithub** [reply]
@merklefruit were you able to integrate Dozer? are you using it currently in your project?

**Reply by merklefruit** [reply]
I added the support for the block number in traces, but I've had other commitments so haven't spent too much time on it so far. I've uploaded the changes [here](https://github.com/merklefruit/dozer/pull/1) if you wanna take a look.

By the way polygon, optimism and arbitrum all work as expected provided you have archive nodes for all of them :)

**Reply by abhishekmishragithub** [reply]
> I added the support for the block number in traces, but I've had other commitments so haven't spent too much time on it so far. I've uploaded the changes [here](https://github.com/merklefruit/dozer/pull/1) if you wanna take a look.
> 
> By the way polygon, optimism and arbitrum all work as expected provided you have archive nodes for all of them :)

Perfect! let us know if you need any help from us :)

**Reply by tomas-dozer** [reply]
Hey @merklefruit ! How are you? I am Tomas and I am working on the new website of Dozer. I am reaching out because I would like to understand the usage that you believe our tool can have. Would you mind to have a quick chat next week to discuss your implementation? Best wishes!
